doi/src/main/java/ca/nrc/cadc/doi/DoiAction.java

```diff
@@ -146,17 +181,18 @@ protected void init(boolean authorize)
 
         // get calling subject
         callingSubject = AuthenticationUtil.getCurrentSubject();
-        if (authorize) {
-            authorizeUser(callingSubject);
-        }
         logInfo.setSubject(callingSubject);
 
         parsePath();
 
         ACIdentityManager acIdentMgr = new ACIdentityManager();
         this.callersNumericId = (Long) acIdentMgr.toOwner(callingSubject);
         this.vospaceDoiClient = new VospaceDoiClient(vaultResourceID, parentPath,
-                callingSubject, includePublic);
+                callingSubject, includePublic, publisherGroupURI, gmsResourceID);
+
+        if (authorize) {
```

Comment:
I didn't explain this very well last time. The init() should not take a parameter. authorizeUser can just be authorize, also not taking a parameter, and authorizeUser can use the callSubject instance variable since authorize() will always be called after the init() method. Then each action can call, or not call, authorize() and then do the action specific checks.